src/chainstate/burn/operations/leader_block_commit.rs

```diff
@@ -135,180 +137,211 @@ where
             block_header_hash: data.block_header_hash,
             new_seed: data.new_seed,
-            parent_block_backptr: data.parent_block_backptr,
+            parent_block_ptr: data.parent_block_ptr,
             parent_vtxindex: data.parent_vtxindex,
-            key_block_backptr: data.key_block_backptr,
+            key_block_ptr: data.key_block_ptr,
             key_vtxindex: data.key_vtxindex,
-            epoch_num: data.epoch_num,
             memo: data.memo,
 
             burn_fee: burn_fee,
-            input: tx.inputs[0].clone(),
+            input: inputs[0].clone(),
 
-            op: OPCODE,
-            txid: tx.txid.clone(),
-            vtxindex: tx.vtxindex,
-            block_number: block_height,
+            txid: tx.txid(),
+            vtxindex: tx.vtxindex(),
+            block_height: block_height,
             burn_header_hash: block_hash.clone(),
-
-            _phantom: PhantomData
         })
     }
 }
 
-impl<A, K> BlockstackOperation<A, K> for LeaderBlockCommitOp<A, K> 
-where
-    A: Address,
-    K: PublicKey
-{
-    fn from_tx(block_height: u64, block_hash: &BurnchainHeaderHash, tx: &BurnchainTransaction<A, K>) -> Result<LeaderBlockCommitOp<A, K>, op_error> {
-        LeaderBlockCommitOp::<A, K>::parse_from_tx(block_height, block_hash, tx)
+impl BlockstackOperation for LeaderBlockCommitOp {
+    fn from_tx(block_header: &BurnchainBlockHeader, tx: &BurnchainTransaction) -> Result<LeaderBlockCommitOp, op_error> {
+        LeaderBlockCommitOp::parse_from_tx(block_header.block_height, &block_header.block_hash, tx)
     }
         
-    fn check(&self, burnchain: &Burnchain, conn: &DBConn) -> Result<CheckResult, op_error> {
-        let leader_key_block_height = self.block_number - (self.key_block_backptr as u64);
-        let parent_block_height = self.block_number - (self.parent_block_backptr as u64);
+    fn check<'a>(&self, burnchain: &Burnchain, block_header: &BurnchainBlockHeader, tx: &mut BurnDBTx<'a>) -> Result<(), op_error> {
+        let leader_key_block_height = self.key_block_ptr as u64;
+        let parent_block_height = self.parent_block_ptr as u64;
+        
+        // this will be the chain tip we're building on
+        let chain_tip = BurnDB::get_block_snapshot(tx, &block_header.parent_block_hash)
+            .expect("FATAL: failed to query parent block snapshot")
+            .expect("FATAL: no parent snapshot in the DB");
+
+        /////////////////////////////////////////////////////////////////////////////////////
+        // There must be a burn
+        /////////////////////////////////////////////////////////////////////////////////////
+        
+        if self.burn_fee == 0 {
+            warn!("Invalid block commit: no burn amount");
+            return Err(op_error::BlockCommitBadInput);
+        }
         
         /////////////////////////////////////////////////////////////////////////////////////
         // This tx's epoch number must match the current epoch
```

Comment:
This epoch check is no longer performed, right? This comment should be removed

Reply:
Correct; will remove.

Reply:
Fixed in 8f8c6e10